pkg/controller/elasticsearch/elasticsearch_controller.go

```diff
@@ -216,6 +216,12 @@ func (r *ReconcileElasticsearch) Reconcile(request reconcile.Request) (reconcile
 
 	state := esreconcile.NewState(es)
 	results := r.internalReconcile(es, state)
+	if results.HasError() {
+		res, err := results.Aggregate()
+		k8s.EmitErrorEvent(r.recorder, err, &es, events.EventReconciliationError, "Reconciliation error: %v", err)
+		return res, err
```

Comment:
Not sure it's OK to not update ES status here. If the status is OK, then on subsequent reconciliations something fails, then the ES status will never reflect that orchestration work may still be pending?